File: Frontend/src/app/pages/model/usuario.ts
import { Residencia } from './residencia';
import { Residente } from './residente';
import { Rol } from './rol';

export interface Usuario {
  id: number;
  user: string;
  password: string;
  fecCreacion: string;
  usuCreacion: string;
  fecModifica: string;
  usuModifica: string;
  idResidencia: Residencia;
  idResidente: Residente;
  idRol: Rol;
}

File: Backend/src/main/java/pe/com/model/Usuario.java
package pe.com.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "usuario")
public class Usuario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column
    private String user;

    @Column
    private String password;

    @Column
    private String fecCreacion;

    @Column
    private String usuCreacion;

    @Column
    private String fecModifica;

    @Column
    private String usuModifica;

    @ManyToOne
    @JoinColumn(name = "id_residencia", nullable=false)
    private Residencia idResidencia;

    @ManyToOne
    @JoinColumn(name = "id_residente", nullable=false)
    private Residente idResidente;

    @ManyToOne
    @JoinColumn(name = "id_rol", nullable=false)
    private Rol idRol;
}
